linode/databasemysql/framework_datasource_schema.go

```diff
@@ -24,6 +25,9 @@ var frameworkDatasourceSchema = schema.Schema{
 		"label": schema.StringAttribute{
 			Description: "A unique, user-defined string referring to the Managed Database.",
 			Computed:    true,
+			Validators: []validator.String{
+				stringvalidator.LengthBetween(3, 32),
```

Comment:
optional: It might also make sense to ensure that the label contains only valid characters :+1:

Reply:
I think RegExp is the way to go, that way we won't need to adjust the function in the future, just the RegExp used. Thoughts on the current implementation?

Reply:
There are multiple set of requirements of label for different product, and a single regex is impossible to serve all of them...